src/librados/librados.cc

```diff
@@ -1266,6 +1266,26 @@ int librados::IoCtx::read(const std::string& oid, bufferlist& bl, size_t len, ui
   return io_ctx_impl->read(obj, bl, len, off);
 }
 
+int librados::IoCtx::repair_read(const std::string& oid, bufferlist& bl,
+				 size_t len, uint64_t off, int flags,
+				 int32_t osdid, epoch_t e)
+{
+  auto c = librados::Rados::aio_create_completion();
```

Comment:
I agree, the first way was duplicating fewer unique things.

In general these librados.cc methods should be doing as little as possible - just the necessities to translate the C++ and C interfaces to a common internal call in IoCtxImpl. You'll notice nothing else in librados.cc calls prepare_assert_ops() for example, all that sort of common code is in IoCtxImpl. This bit can be cleaned up when we add the C interface though, this pr doesn't have to do everything at once.